numpy/ma/core.py

```diff
@@ -82,6 +82,9 @@
 MaskType = np.bool_
 nomask = MaskType(0)
 
+class MaskedArrayFutureWarning(FutureWarning):
```

Comment:
Would you like me to change the inheritance then or drop it completely?

The point was to make sure that this special case of `MaskedArrayFutureWarning` could be dealt with separately from `FutureWarning`. So one would not need to filter out all `FutureWarning`s, but could just filter out `MaskedArrayFutureWarning`s. Also one could just as easily promote only `MaskedArrayFutureWarning` to errors to check and see where they are assuming the old behavior, but not correcting for it.